src/dbus/server/dbus_agent.cpp

```diff
@@ -52,8 +54,13 @@ otbrError DBusAgent::Init(void)
     std::string serverName = OTBR_DBUS_SERVER_PREFIX + mInterfaceName;
 
     dbus_error_init(&dbusError);
-    DBusConnection *conn = dbus_bus_get(DBUS_BUS_SYSTEM, &dbusError);
-    mConnection          = std::unique_ptr<DBusConnection, std::function<void(DBusConnection *)>>(
+    DBusConnection *conn                = nullptr;
+    auto            connection_deadline = std::chrono::steady_clock::now() + std::chrono::seconds(10);
+    while (conn == nullptr && std::chrono::steady_clock::now() < connection_deadline)
+    {
+        conn = dbus_bus_get(DBUS_BUS_SYSTEM, &dbusError);
```

Comment:
need to print a WARN message if conn is nullptr?

Reply:
Done.